### PR TITLE
dmirano/3282 apd overview cypress test

### LINKED ISSUE #3282: [Feature] Create e2e test for APD Overview
**Description and related issues -**
Write a test that creates an APD and fills in all of the fields on the APD Overview page. Verify that those values are the same in the Export and Submit view.

### Test cases needed…

- [x] should be able to log in
- [x] should be able to create an APD
- [x] should have default values in the export view
    - [x] verify that Program introduction, HIT overview, HIE overview, and MMIS overview says "No response was provided" when those fields are empty
- [x] should be able to fill out the APD Overview
    - [x] should be able to choose FFYs
        - [x] verify that the first 2 years are are checked
        - [x] verify that you can check the unchecked year
        - [x] verify that the FFY in the header updates to include the new year checked
        - [x] verify that the header shows Saving and then Saved
        - [x] verify that when you uncheck the year you just check the delete confirmation modal pops up
    - [x] should be able to click Cancel on the delete confirmation modal
        - [x] verify that clicking Cancel closes the modal and leaves the year checked
    - [x] should be able to click Delete FFY on the delete confirmation modal
        - [x] verify that clicking Delete FFY closes the modal and unchecks the year
    - [x] should be able to enter Introduction
    - [x] should be able to enter HIT Overview
    - [x] should be able to enter HIE Overview
    - [x] should be able to enter MMIS Overview
    - [x] should be able to click Continue
        - [x] verify that you are taken to the Key State Personnel page
- [x] should have the same data in the export view
    - [x] verify that FFY range in the header matches what you check
    - [x] verify that Program introduction, HIT overview, HIE overview, and MMIS overview have the values that you entered above

### PR DESCRIPTION
**Pull request by mirano-darren**
resolves #3282

**Description-**
Cypress test for filling out the APD overview section. 

**This pull request changes...**

- adds cypress dependencies in /web/package.json
- cypress folder in /web/

**This pull request also touches…**

- hopefully nothing other than the web directory

**This pull request was tested in the follow ways…**

- tested off a fresh checkout (wasn't too sure what to put here)

**Steps to manually verify this change...**

0. rerun npm install in /web/ and /api/
1. build docker containers
2. add a "cypress.env.json" file in the web directory. This file is in Keybase
3. run app as normal
4. launch cypress in the /web/ directory with "npx cypress open"
5. click on the test "apd-overview.spec.js"
6. cypress should automatically open and run the tests
7. all three tests should pass

### This pull request is ready to review when...

- [X] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [X] Code has been reviewed by someone other than the original author
- [X] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: web/cypress/integration/apd-tests/apd-overview.spec.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    cy.contains('Back').click();
+  });
+
+  it('adds value to the APD', function () {
+    //Gets list of available years
+    cy.get("[class='ds-c-choice']").each(($el, index, list) => {
+      years.push(list[index].value);
+    });
+
+    cy.then(() => {
```
[the remark this answers is not in the document]
Comment (thetif) [reply web/cypress/integration/apd-tests/apd-overview.spec.js:65]:
```
cy.get("[class='ds-c-choice']").each(($el, index, list) => {
    years.push(list[index].value);
    if (!list[index].checked) {
      cy.findByRole('checkbox', { name: list[index].value }).check({
        force: true
      });

      cy.contains('Saving').should('exist');
      cy.contains('Saved').should('exist');
    }
  });
  cy.then(() => {
    cy.get('[class="ds-h1 apd--title"]').should('contain', years[0]);
    cy.get('[class="ds-h1 apd--title"]').should(
      'contain',
      years[years.length - 1]
    );
  });
```

**Review by thetif — APPROVED** [review]
Great job! Just a couple of suggestions of things that you could refactor.

File: web/cypress/integration/apd/01-apd-overview.spec.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    //Tests continue button
+    cy.contains('Continue').click();
+    cy.get('[class="ds-h2"]').should('contain', 'Key State Personnel');
+    cy.contains('Back').click();
+  });
+
+  it('adds value to the APD', function () {
+    //Gets list of available years
+    cy.get("[class='ds-c-choice']").each(($el, index, list) => {
```
Comment (thetif) [inline web/cypress/integration/apd/01-apd-overview.spec.js:61]: try doing this and the next loop as one loop
```
cy.get("[class='ds-c-choice']").each(($el, index, list) => {
    years.push(list[index].value);
    if (!list[index].checked) {
      cy.findByRole('checkbox', { name: list[index].value }).check({
        force: true
      });

      cy.contains('Saving').should('exist');
      cy.contains('Saved').should('exist');
    }
  });
  cy.then(() => {
    cy.get('[class="ds-h1 apd--title"]').should('contain', years[0]);
    cy.get('[class="ds-h1 apd--title"]').should(
      'contain',
      years[years.length - 1]
    );
  });
  ```

File: web/cypress/integration/apd/apd-activities/activities.spec.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
+/// <reference types="cypress" />
```
Comment (thetif) [inline web/cypress/integration/apd/apd-activities/activities.spec.js:1]: can you remove this file from this branch and just work on it in your new branch?

**Review by thetif — CHANGES_REQUESTED** [review]

**Review by tbolt — APPROVED** [review]
Nice work!

**Comment by thetif** [comment]
I'm getting one test failure, can you check it out?
Screen Shot 2021-07-28 at 12 43 00 PM

**Review by thetif — APPROVED** [review]

**Comment by thetif** [comment]
all tests passing, good to merge 

Screen Shot 2021-07-28 at 2 47 48 PM

**Pull request merged**